### PR TITLE
Add subspace orthogonality analysis for factored processes

### PR DESCRIPTION
**Pull request by loren-ac**
## Summary

Adds functionality to compute orthogonality metrics between activation
subspaces when models are trained on factored processes (processes whose
belief state is a Cartesian product of subprocess belief states).

## Key Features

- **Subspace orthogonality computation**: Computes pairwise orthogonality
metrics between learned coefficient subspaces using QR decomposition and
SVD
- **Orthogonality metrics**:
  - Subspace overlap (0 = fully orthogonal, 1 = fully aligned / contained)
  - Singular values of interaction matrix between orthonormal bases
  - Spectral norm, participation ratio, entropy, effective rank
- **Efficient computation**: Computes coefficients once, reuses for both
regression and orthogonality
- **Clean API**: Separate `coeffs` (linear transformation) and `intercept`
 (translation) in return structure

#### `linear_regression.py`:
- Add `compute_subspace_orthogonality` parameter to 
`layer_linear_regression()`
- Implement `_compute_all_pairwise_orthogonality()` and `_compute_subspace_orthogonality()` for pairwise subspace metrics for multi-factor scenarios
- Ensure orthogonality computed only from coefficients (intercepts
excluded)
- Unify SVD functionality via `use_svd` flag

#### `layerwise_analysis.py`:
- Update validators to accept new parameters: `concat_belief_states`, `compute_subspace_orthogonality`, `use_svd`
- Ensure proper parameter forwarding from analysis framework to regression functions
- Maintain backward compatibility with existing analyses

### API improvements:
- Remove `to_factors` flag and introduce `concat_belief_states` flag 
- Use `concat_belief_states` to determine whether to do regression on concatenated beliefs vs factored beliefs
- Rename `projections` dictionary to `arrays` (more descriptive)
- Store linear regression parameters in `arrays` as `coeffs` and `intercept`
- Omit `intercept` key when `fit_intercept=False`

#### Orthogonality computation tests (`test_linear_regression.py`):
- Updates existing tests to work with parameter/API changes
- 9 comprehensive orthogonality tests covering:
  - Orthogonal, aligned, and hierarchically-contained subspaces
  - Multi-factor pairwise combinations
  - Different subspace dimensions
  - Integration with SVD regression
  - Edge cases and default behavior
- Principled numerical thresholds based on machine precision
- Informative assertion messages for debugging

#### Parameter validation tests (`test_layerwise_analysis.py`):
- Validates that the layerwise analysis framework correctly accepts and processes the new parameters ( `concat_belief_states`, `compute_subspace_orthogonality`, `use_svd`)
- Ensures validators properly handle defaults, type checking, and parameter forwarding
- Tests updated to verify the new return structure (separate `coeffs`/`intercept` keys)

---

> [!NOTE]
> Adds pairwise subspace-orthogonality metrics for factored beliefs, separates coeffs/intercept in outputs, and unifies linear regression/SVD via a use_svd flag with updated validators and tests.
> 
> - **Analysis (linear regression)**:
>   - Compute pairwise subspace-orthogonality metrics between factor coefficient subspaces (overlap, singular values, participation ratio, entropy, effective rank).
>   - Return arrays with separated `coeffs` and `intercept` (omit `intercept` when `fit_intercept=False`).
>   - Support multi-factor flows: per-factor regression by default; optional `concat_belief_states` to fit jointly and split back; reuse params for metrics.
>   - Unify solvers via `use_svd` (with optional `rcond_values`); expose best rcond appropriately (concat vs per-factor).
> - **Layerwise orchestration/validation**:
>   - Update validators to accept `concat_belief_states`, `compute_subspace_orthogonality`, `use_svd`, `rcond_values`; enforce constraints and defaults.
>   - Register `linear_regression_svd` via `partial(layer_linear_regression, use_svd=True)` and guard `use_svd` in validator.
> - **API changes**:
>   - Replace `to_factors` with `concat_belief_states`.
>   - Projections output now includes parameter arrays (`projected`, `coeffs`, `intercept`).
> - **Tests**:
>   - Extensive new/updated tests covering orthogonality scenarios, concat vs separate equivalence, SVD parity, validator behavior, and new return structure.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 51283f738330bf8d229ae1ad4f03df80502faae7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: simplexity/analysis/linear_regression.py [hunk 1]
```diff
@@ -128,67 +170,348 @@ def linear_regression_svd(
...
+
+
+def _compute_subspace_orthogonality(
+    coeffs_pair: list[jax.Array],
+) -> tuple[dict[str, float], dict[str, jax.Array]]:
+    """Compute orthogonality metrics between two coefficient subspaces.
+
+    Args:
+        coeffs_pair: List of two coefficient matrices (excludes intercept)
+    """
```
[the remark this answers is not in the document]
Comment (loren-ac) [reply simplexity/analysis/linear_regression.py:322]: I'm currently working on this

**Review by ealt — APPROVED** [review]
